src/uast.cc

```diff
@@ -180,14 +182,24 @@ Nodes *UastFilter(const Uast *ctx, void *node, const char *query) {
   results = result->nodeTab;
   size = (result) ? result->nodeNr : 0;
 
-  if (NodesSetSize(nodes, size) != 0) {
+  // Sometimes libxml return null results for some invalid queries; do
+  // a first pass to discard them and get the real size for the result
```

Comment:
Is this behaviour documented somewhere? Any idea of why libxml behaves like this?

Reply:
Didn't find anything, and actually on all tests except the file that Vadim's used in the issue it works (in the sense that it returns an error), but not for this specific one, I wan't to create a smaller test case but any effort until now to reduce the test just changes the behavior to the expected one (I'll keep trying).